subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/converters/IdeaModuleDependencyTargetNameMixin.java

```diff
@@ -29,7 +29,7 @@ public IdeaModuleDependencyTargetNameMixin(IdeaDependency ideaModuleDependency)
     public String getTargetModuleName() {
         if (ideaModuleDependency instanceof IdeaModuleDependency) {
             IdeaModuleDependency dependency = (IdeaModuleDependency) ideaModuleDependency;
-            return dependency.getDependencyModule().getName();
+            return dependency.getTargetModuleName();
```

Comment:
This doesn't work for target Gradle versions < 3.1.  I've tried various solutions to get around this.  I don't think there is anything I can do, short of some very shady reflections and accessibility hacks, to be able to determine the module name.  

The dependency module is still there... it shows up in the debugger.  But there's no way to access it, as far as I can tell.  

@adammurdoch, any ideas?